pkg/scheduler/connections/pool.go

```diff
@@ -0,0 +1,155 @@
+package connections
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"sync"
+	"time"
+
+	schedulerv1pb "github.com/dapr/dapr/pkg/proto/scheduler/v1"
+	"github.com/dapr/dapr/pkg/scheduler"
+	"github.com/dapr/kit/logger"
+)
+
+var log = logger.NewLogger("dapr.runtime.scheduler")
+
+// Pool represents a connection pool for namespace/appID separation of sidecars to schedulers.
+type Pool struct {
```

Comment:
Do not export.

Reply:
as discussed, this will mv to something like: pkg/scheduler/server/internal/pool.go